hs_app_timeseries/tests/test_timeseries_metadata.py

```diff
@@ -1,50 +1,57 @@
-from hs_core.hydroshare import utils, users, resource
-from hs_core.models import BaseResource, Creator, Contributor, CoreMetaData, \
-    Coverage, Rights, Title, Language, Publisher, Identifier, \
-    Type, Subject, Description, Date, Format, Relation, Source
+from dateutil import parser
 
-from hs_app_timeseries.models import *
+from xml.etree import ElementTree as ET
 
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import UploadedFile
 from django.contrib.auth.models import Group, User
+
 from hs_core import hydroshare
-from dateutil import parser
-from lxml import etree
+from hs_core.hydroshare import utils, resource
+from hs_core.models import BaseResource, CoreMetaData, Creator, Contributor, Coverage, Rights, Title, Language, \
+    Publisher, Identifier, Type, Subject, Description, Date, Format, Relation, Source
+from hs_core.testing import MockIRODSTestCaseMixin
+from hs_app_timeseries.models import TimeSeriesResource, Site, Variable, Method, ProcessingLevel, TimeSeriesResult
 
 
-class TestTimeSeriesMetaData(TestCase):
+class TestTimeSeriesMetaData(MockIRODSTestCaseMixin, TestCase):
     def setUp(self):
-        try:
-            #self.user = User.objects.create_user('user1', email='user1@nowhere.com')
-            self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
-            self.user = hydroshare.create_account(
-                'user1@nowhere.com',
-                username='user1',
-                first_name='Creator_FirstName',
-                last_name='Creator_LastName',
-                superuser=False,
-                groups=[self.group]
-            )
-        except:
-            self.tearDown()
-            self.user = User.objects.create_user('user1', email='user1@nowhere.com')
-
+        super(TestTimeSeriesMetaData, self).setUp()
+        self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        self.user = hydroshare.create_account(
+            'user1@nowhere.com',
+            username='user1',
+            first_name='Creator_FirstName',
+            last_name='Creator_LastName',
+            superuser=False,
+            groups=[self.group]
+        )
 
         self.resTimeSeries = hydroshare.create_resource(
             resource_type='TimeSeriesResource',
             owner=self.user,
-            title='Test Time Series Resource',
-            keywords=['kw1', 'kw2']
+            title='Test Time Series Resource'
         )
+        self.odm2_sqlite_file_name = 'ODM2_valid.sqlite'
+        self.odm2_sqlite_file = 'hs_app_timeseries/tests/{}'.format(self.odm2_sqlite_file_name)
+        self.odm2_sqlite_file_obj = open(self.odm2_sqlite_file, 'r')
+        self.odm2_sqlite_bad_file_name = 'ODM2_invalid.sqlite'
+        self.odm2_sqlite_bad_file = 'hs_app_timeseries/tests/{}'.format(self.odm2_sqlite_bad_file_name)
+        self.odm2_sqlite_bad_file_obj = open(self.odm2_sqlite_bad_file, 'r')
+
+        text_file_name = 'ODM2.txt'
```

Comment:
see the `tempfile` module for help finding a temporary location for all these files. We don't want to just be populating the local directory writing files willy nilly.